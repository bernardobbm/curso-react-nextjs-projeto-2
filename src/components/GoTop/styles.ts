import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    background-color: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 2.5rem;
    height: 2.5rem;
    border-radius: 50%;

    position: fixed;
    bottom: 1.25rem;
    right: 1.25rem;
    z-index: 6;
  `}
`;
