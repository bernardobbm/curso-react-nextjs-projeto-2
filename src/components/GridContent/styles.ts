import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    text-align: center;
    max-width: 580px;
    margin: 0 auto;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacing.large} 0;
  `}
`;
