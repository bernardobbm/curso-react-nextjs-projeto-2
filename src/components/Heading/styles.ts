import styled, { css } from 'styled-components';
import { HeadingProps } from '.';

export const Title = styled.h1<HeadingProps>`
  ${({ theme, light }) => css`
    color: ${light ? theme.colors.white : theme.colors.primaryColor};
  `}
`;
