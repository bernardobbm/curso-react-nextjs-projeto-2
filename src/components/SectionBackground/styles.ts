import styled, { css, DefaultTheme } from 'styled-components';
import { SectionBackgroundProps } from '.';

function containerBackgroudActivate(theme: DefaultTheme) {
  return css`
    background-color: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
  `;
}

export const Container = styled.div<SectionBackgroundProps>`
  ${({ theme, background }) => css`
    background-color: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    min-height: 100vh;
    display: flex;
    align-items: center;

    ${background && containerBackgroudActivate(theme)}
  `}
`;
