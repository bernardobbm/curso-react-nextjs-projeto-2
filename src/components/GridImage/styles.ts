import styled, { css } from 'styled-components';

import { Container as TextComponentContainerStyle } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponentContainerStyle} {
      margin-bottom: ${theme.spacing.xhuge};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${() => css`
    width: 100%;
    transition: all 0.3s ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
