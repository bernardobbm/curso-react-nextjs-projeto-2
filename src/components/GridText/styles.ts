import styled, { css } from 'styled-components';

import { Title as HeadingTitleStyle } from '../Heading/styles';
import { Container as TextComponentContainerStyle } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponentContainerStyle} {
      margin-bottom: ${theme.spacing.xhuge};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.large};
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    ${HeadingTitleStyle} {
      position: relative;
      left: 3.125rem;
    }

    ${HeadingTitleStyle}:before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 4.5rem;
      top: -2rem;
      left: -3.8rem;
      transform: rotate(8deg);
    }
  `}
`;
