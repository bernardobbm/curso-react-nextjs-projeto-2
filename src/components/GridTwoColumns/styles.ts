import styled, { css } from 'styled-components';

import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacing.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }

    ${Title} {
      margin-bottom: ${theme.spacing.large};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacing.large};
      text-align: center;
    }
  `}
`;

export const ImageContainer = styled.div`
  ${() => css``}
`;

export const Image = styled.img`
  ${() => css``}
`;
