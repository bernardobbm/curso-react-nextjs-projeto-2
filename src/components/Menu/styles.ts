import styled, { css } from 'styled-components';

import { Container as SectionContainerStyle } from '../SectionContainer/styles';
import { Title as HeadingTitleStyle } from '../Heading/styles';

interface StyleProps {
  visible: boolean;
}

function menuVisible() {
  return css`
    visibility: visible;
    opacity: 1;
  `;
}

export const Container = styled.div<StyleProps>`
  ${({ theme, visible }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    border-bottom: ${theme.colors.mediumGray};
    background-color: ${theme.colors.white};
    transition: all 0.3s ease-in-out;

    & ${SectionContainerStyle} {
      padding-top: 0;
      padding-bottom: 0;
    }

    & ${HeadingTitleStyle} {
      margin-top: 0;
      margin-bottom: 0;
    }

    @media ${theme.media.lteMedium} {
      height: 100vh;
      visibility: hidden;
      opacity: 0;

      ${visible && menuVisible()}

      & ${SectionContainerStyle} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        overflow-y: auto;
      }

      ${HeadingTitleStyle} {
        padding-bottom: ${theme.spacing.xxlarge};
        display: flex;
        justify-content: center;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;

    @media ${theme.media.lteMedium} {
      display: block;
      text-align: center;
      padding: ${theme.spacing.xxlarge} 0;
    }
  `}
`;

export const Button = styled.button<StyleProps>`
  ${({ theme, visible }) => css`
    z-index: 6;
    position: fixed;
    top: 1.25rem;
    right: 1.25rem;
    width: 2rem;
    height: 2rem;
    background-color: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    cursor: pointer;
    border: none;
    display: none;
    pointer-events: ${visible ? 'none' : 'all'};

    @media ${theme.media.lteMedium} {
      display: flex;
      align-items: center;
      justify-content: center;
    }
  `}
`;
