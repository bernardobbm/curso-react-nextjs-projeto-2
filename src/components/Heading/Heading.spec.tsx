import { screen } from '@testing-library/react';
import { describe, it } from 'vitest';

import { Heading } from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

describe('<Heading />', () => {
  it('should render with default values', () => {
    renderTheme(<Heading>texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'texto' });

    expect(heading).toHaveStyle({
      color: theme.colors.primaryColor,
      'font-size': theme.font.size.xhuge,
      'text-transform': 'none',
    });
  });

  it('should render with white color', () => {
    renderTheme(<Heading colorDark={false}>texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'texto' });

    expect(heading).toHaveStyle({
      color: theme.colors.white,
    });
  });

  it('should render with medium heading size', () => {
    renderTheme(<Heading size="small">texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'texto' });

    expect(heading).toHaveStyle({
      'font-size': theme.font.size.medium,
    });
  });

  it('should render with big heading size', () => {
    renderTheme(<Heading size="big">texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'texto' });

    expect(heading).toHaveStyle({
      'font-size': theme.font.size.xlarge,
    });
  });

  it('should render with medium heading size', () => {
    renderTheme(<Heading size="medium">texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'texto' });

    expect(heading).toHaveStyle({
      'font-size': theme.font.size.large,
    });
  });

  it('should render with huge heading size', () => {
    renderTheme(<Heading size="huge">texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'texto' });

    expect(heading).toHaveStyle({
      'font-size': theme.font.size.xhuge,
    });
  });

  it('should render correct font-size when screen smallen than 768px', () => {
    renderTheme(<Heading size="huge">texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'texto' });

    expect(heading).toHaveStyleRule('font-size', theme.font.size.xlarge, {
      media: theme.media.lteMedium,
    });
  });

  it('should render with uppercase letters', () => {
    renderTheme(<Heading uppercase>texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'texto' });

    expect(heading).toHaveStyle({
      'text-transform': 'uppercase',
    });
  });

  it('should render with correct heading element', () => {
    const { container } = renderTheme(<Heading as="h6">texto</Heading>);
    const h6 = container.querySelector('h6');

    expect(h6?.tagName.toLowerCase()).toBe('h6');
  });
});
