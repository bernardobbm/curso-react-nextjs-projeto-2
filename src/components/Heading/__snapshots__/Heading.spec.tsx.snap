// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Heading /> > should match to snapshot 1`] = `
.c0 {
  color: #0a1128;
  font-size: 4rem;
  text-transform: none;
}

@media (max-width:768px) {
  .c0 {
    font-size: 2.5rem;
  }
}

<h1
  class="c0"
>
  texto
</h1>
`;
